packages/network/src/Network.ts

```diff
@@ -1,250 +0,0 @@
-  // `onProgress` if you provide it. The guaranteed call is the one at the end, where the value is 1.
-  onProgress?: (fractionCompleted: number) => void,
-  offset = 0
-): Promise<T[]> => {
-  const promises: Promise<T[]>[] = [];
-  let loadedSoFar = 0;
-
-  for (let page = 0; page * querySize + offset < total; page += 1) {
-    const start = page * querySize + offset;
-    const end = Math.min((page + 1) * querySize + offset, total);
-    const loadedThisBatch = end - start;
-    promises.push(
-      new Promise<T[]>(async (resolve) => {
-        let res: T[] = [];
-        while (res.length === 0) {
-          res = await getterFn(start, end);
-          loadedSoFar += loadedThisBatch;
-          onProgress && onProgress(loadedSoFar / total);
-        }
-
-        resolve(res);
-      })
-    );
-  }
-
-  const unflattenedResults = await Promise.all(promises);
-
-  onProgress && onProgress(1);
-
-  return unflattenedResults.flat();
-};
-
-/**
- * Given a transaction hash and a JsonRpcProvider, waits for the given transaction to complete.
- */
-export function waitForTransaction(
-  provider: providers.JsonRpcProvider,
-  txHash: string
-): Promise<providers.TransactionReceipt> {
-  return retry(
-    async (tries) => {
-      console.log(`[wait-tx] WAITING ON tx hash: ${txHash} tries ${tries}`);
-
-      try {
-        const receipt = await timeout(provider.getTransactionReceipt(txHash), 30 * 1000);
-
-        if (receipt) {
-          console.log(`[wait-tx] FINISHED tx hash: ${txHash} tries ${tries}`);
-          return receipt;
-        } else {
-          return Promise.reject(new Error("couldn't get receipt"));
-        }
-      } catch (e) {
-        console.error(`[wait-tx] TIMED OUT tx hash: ${txHash} tries ${tries} error:`, e);
-        return Promise.reject(e);
-      }
-    },
-    {
-      // TODO: Should we set maxRetryTime?
-      retries: DEFAULT_MAX_CALL_RETRIES,
-      minTimeout: 2000,
-      maxTimeout: 60_000,
-      factor: 1.5,
-      onFailedAttempt(e) {
-        console.log(`[wait-tx] SLEEPING tx hash: ${txHash} tries ${e.attemptNumber} sleeping...`);
-      },
-    }
-  );
-}
-
-/**
- * @param contractAddress the address of the contract you want to connect to
- * @param contractABI a javacript object representing the ABI
- */
-export function createContract<C extends Contract>(
-  contractAddress: string,
-  contractABI: ContractInterface,
-  provider: providers.JsonRpcProvider,
-  signer?: Wallet
-): C {
-  return new Contract(contractAddress, contractABI, signer ?? provider) as C;
-}
-
-/**
- * Creates a new {@link JsonRpcProvider}, and makes sure that it's connected to xDai if we're in
- * production.
- */
-export function makeProvider(rpcUrl: string): providers.JsonRpcProvider {
-  let provider;
-
-  if (rpcUrl.startsWith('wss://')) {
-    provider = new providers.WebSocketProvider(rpcUrl);
-  } else {
-    provider = new providers.StaticJsonRpcProvider(rpcUrl);
-    provider.pollingInterval = 1200;
```

Comment:
just flagging: I changed the pollingInterval in this package vs. DF's polling

Reply:
Good catch. I'll update to use 1000 since we're no longer batching event filters. I will also consider a constructor argument but I'm less inclined for that.